cartography/intel/semgrep/findings.py

```diff
@@ -206,6 +221,7 @@ def sync(
     vulns, usages = transform_sca_vulns(raw_vulns)
     load_semgrep_sca_vulns(neo4j_sesion, vulns, semgrep_deployment["id"], update_tag)
     load_semgrep_sca_usages(neo4j_sesion, usages, semgrep_deployment["id"], update_tag)
+    run_analysis_job('semgrep_sca_risk_analysis.json', neo4j_sesion, common_job_parameters)
```

Comment:
One thing to keep in mind: this analysis job is currently not scoped to run for a specific Semgrep tenant or specific Github tenant, so the statements where it deletes properties will happen for _all_ Github tenants and _all_ Semgrep tenants.

I think we should

1. edit the analysis job so that we only set or remove properties for Semgrep objects in the current Semgrep tenant.
2. introduce the idea of a scoped analysis job where we can define a set of analysis queries to run fora given tenant.


For item 2 we would make the following updates:

I. cartography/util.py:

```python
def run_scoped_analysis_job(
    filename: str,
    neo4j_session: neo4j.Session,
    common_job_parameters: Dict,
) -> None:
    """
    Enriches existing graph data scoped to a given sub resource - e.g. the current AWS account.
    Runs the queries in the cartography.data.jobs.scoped_analysis directory for the given `filename`. View the queries
    in cartography.data.jobs.scoped_analysis for specifics.
    """
    GraphJob.run_from_json(
        neo4j_session,
        read_text('cartography.data.jobs.scoped_analysis', filename),
        common_job_parameters,
        get_job_shortname(filename),
    )
```

II. add cartography/data/jobs/scoped_analysis/ as a new folder and move the semgrep_sca_risk_analysis.json file to it.

I played with this idea before but [never actually merged it](https://github.com/lyft/cartography/pull/1219/commits/3c7b0bd25b0798b71c00d48be0f970dd8c118031#diff-24cb7d746e097729d20949fb290b122181beebc03590e9ef937d2519614f32b8).

---

For the sake of giving an complete explanation, I'm going to write more details here but I do not think we should build it this way. Feel free to ignore if you're trying to just get on with the PR.

Now, if having this analysis job run for all tenants is the intended behavior, then we can have it the way you had it because all cartography analysis jobs eventually get run at the end of the sync (the [call itself](https://github.com/lyft/cartography/blob/2fa2cb0aa54b74844d9711c8bd30f759bf315f9f/cartography/sync.py#L56), the [details](https://github.com/lyft/cartography/blob/2fa2cb0aa54b74844d9711c8bd30f759bf315f9f/cartography/intel/analysis.py#L30-L31)). I actually don't think we should do it this way though because if you mix and match sync stages as part of a [custom sync](https://lyft.github.io/cartography/dev/developer-guide.html?highlight=custom%20sync#implementing-custom-sync-commands) you'll have to end up writing an internal shim to run the analysis job, so this can be messy.

Reply:
This is actually interesting, I was assuming a single tenant but it can clearly apply to multiple ones and the analysis wouldn't change (unless we add an external piece of information for the analysis).

So, we can apply the same rules to each Semgrep deployment and shouldn't affect the state of the graph.

One extra piece of information is that we query the deployment based on the `SEMGREP_APP_TOKEN`. It always returns a single tenant based on their current documentation: https://semgrep.dev/api/v1/docs/#tag/Deployment/operation/semgrep_app.saas.handlers.deployment.openapi_list_deployments
> Currently available auth scope does not extend over more than 1 deployment. This endpoint returns that single deployment your token can access. The endpoint additionally returns links towards related resources available on this web API.

So, I will go with option 1, considering currently it will:
1. Always be a single deployment per sync
2. Use the deployment to only update the `reachability_risk` property for `(:SemgrepDeployment{id:DEPLOYMENT_D. lastupdated: UPDATE_TAG})-[:RESOURCE]->(:SCASemgrepFinding)` nodes.